### ISSUE TITLE
Performance regression in 1.12.10

### ISSUE THREAD

**Issue by davidtheclark** [webpack/webpack#1860]
I noticed a very significant slowdown in build times when upgrading from 1.12.9 to 1.12.10. On my initial build, the difference is not huge; but on incremental rebuilds when watching, the time went from ~1s to ~7.5s -- which *is* huge.

With 1.12.9, I was getting builds like this:

```
Time: 11420ms
chunk    {0} docs.js (docs) 1.09 MB {3} [rendered]
chunk    {1} 1.1.js 337 kB {0} {3} [rendered]
chunk    {2} 2.2.js 34.4 kB {0} {3} [rendered]
chunk    {3} smartui.js (smartui) 517 kB {4} [rendered]
chunk    {4} vendor.js (vendor) 1.28 MB [rendered]
...
Time: 1193ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
[BS] Reloading Browsers...
Time: 1053ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
...
Time: 1123ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
...
```

After upgrading to 1.12.10 and changing nothing else, I get builds like this:

```
Time: 15030ms
chunk    {0} docs.js (docs) 1.09 MB {3} [rendered]
chunk    {1} 1.1.js 337 kB {0} {3} [rendered]
chunk    {2} 2.2.js 34.4 kB {0} {3} [rendered]
chunk    {3} smartui.js (smartui) 517 kB {4} [rendered]
chunk    {4} vendor.js (vendor) 1.28 MB [rendered]
...
Time: 7616ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
...
Time: 7457ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
...
Time: 7197ms
chunk    {0} docs.js (docs) 1.09 MB {3}
chunk    {1} 1.1.js 337 kB {0} {3}
chunk    {2} 2.2.js 34.4 kB {0} {3}
chunk    {3} smartui.js (smartui) 517 kB {4}
chunk    {4} vendor.js (vendor) 1.28 MB
...
```

I double-checked this by switching versions back-and-forth a few times, and got the same performance difference each time.

Is there any other information you would like to provide clues as to what is going on?

**Comment by bebraw** [comment]
Here are the [changes](https://github.com/webpack/webpack/compare/v1.12.9...v1.12.10) between the versions. Maybe that helps a bit. If you have time, you could try to run against specific commits. That would catch the one that caused the regression. After that we need to understand why.

**Comment by mistadikay** [comment]
same here, updating build became awfully slow

**Comment by davidtheclark** [comment]
Looks like the first commit in the upgrade causes it: https://github.com/webpack/webpack/commit/e76c71269075017f535ff13c9e7a000f51b4505f

**Comment by sokra** [comment]
aha... that's unexpected...

**Comment by eddiemonge** [comment]
did you mean 1.12.9 and 1.12.10?

**Comment by davidtheclark** [comment]
@eddiemonge: Yep, I corrected all the text above.

**Comment by chiplay** [comment]
Seeing the same issue with 0.12.10

**Comment by carlos-stored** [comment]
Same problem with 0.12.10

**Comment by carlosrberto** [comment]
Same problem with 0.12.10

**Comment by sokra** [comment]
reverted

**Comment by davidtheclark** [comment]
Is the reversion published in 1.12.11?

**Comment by IngwiePhoenix** [comment]
It is. See the commit log, its right in there. :)
